examples/src/bin/custom_editable.rs

```diff
@@ -0,0 +1,434 @@
+                container: gtk::Box::new(gtk::Orientation::Horizontal, 0),
+                button: RefCell::default(),
+                label: gtk::Label::new(None),
+                has_close_button: Cell::new(false),
+            }
+        }
+    }
+
+    #[glib::object_subclass]
+    impl ObjectSubclass for CustomTag {
+        const NAME: &'static str = "CustomTag";
+        type Type = super::CustomTag;
+        type ParentType = gtk::Widget;
+
+        fn class_init(klass: &mut Self::Class) {
+            klass.set_css_name("tag");
+        }
+    }
+
+    impl ObjectImpl for CustomTag {
+        fn properties() -> &'static [ParamSpec] {
+            static PROPERTIES: Lazy<Vec<ParamSpec>> = Lazy::new(|| {
+                vec![
+                    ParamSpec::string(
+                        "label",
+                        "Label",
+                        "Label",
+                        Some(""),
+                        glib::ParamFlags::READWRITE,
+                    ),
+                    ParamSpec::boolean(
+                        "has-close-button",
+                        "Has close button",
+                        "Whether this tag has a close button",
+                        false,
+                        glib::ParamFlags::READWRITE,
+                    ),
+                ]
+            });
+            PROPERTIES.as_ref()
+        }
+
+        fn get_property(&self, _obj: &Self::Type, _id: usize, pspec: &ParamSpec) -> glib::Value {
+            match pspec.get_name() {
+                "label" => self.label.get_text().to_value(),
+                "has-close-button" => self.has_close_button.get().to_value(),
+                _ => unimplemented!(),
+            }
+        }
+
+        fn set_property(
+            &self,
+            tag: &Self::Type,
+            _id: usize,
+            value: &glib::Value,
+            pspec: &ParamSpec,
+        ) {
+            match pspec.get_name() {
+                "label" => self.label.set_text(value.get().unwrap().unwrap()),
+                "has-close-button" => {
+                    tag.set_has_close_button(value.get_some().unwrap());
+                }
+                _ => unimplemented!(),
+            }
+        }
+
+        fn signals() -> &'static [Signal] {
+            static SIGNALS: Lazy<Vec<Signal>> = Lazy::new(|| {
+                vec![
+                    Signal::builder("closed", &[], <()>::static_type().into()).build(),
+                    Signal::builder("clicked", &[], <()>::static_type().into()).build(),
+                ]
+            });
+            SIGNALS.as_ref()
+        }
+
+        fn constructed(&self, tag: &Self::Type) {
+            self.container.set_parent(tag);
+            self.container.append(&self.label);
+
+            let gesture = gtk::GestureClick::new();
+            gesture.connect_released(clone!(@weak tag => move |_gesture, _n_press, _x, _y| {
+                tag.emit(CustomTag::signals()[1].signal_id(), &[]).unwrap();
```

Comment:
Without a named `const` or something, the `1` here is less understandable than doing `emit_by_name("clicked")`